### PR TITLE
Parse and print `Ppat_open`

### PR DESCRIPTION
**Pull request by anmonteiro**
fixes https://github.com/facebook/reason/issues/2301

### CODE REVIEW

File: formatTest/typeCheckedTests/expected_output/mlSyntax.4.04.0.re [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module Foo = {
+  type t = {name: string};
+};
+
+let foo = (Foo.{name}) => ();
```
Comment (jordwalke) [inline formatTest/typeCheckedTests/expected_output/mlSyntax.4.04.0.re]: Want to add some more tests covering the various places patterns can occur? Destructuring tuples, switch statements etc.

**Comment by anmonteiro** [comment]
@jordwalke I added some more parsing rules to support parsing e.g. `let Foo.[bar] = baz;` and other patterns.

There's only one caveat with the current state of this PR: it parses `let Foo.((bar, baz)) = qux;` but not `let Foo.(bar, baz) = qux;`. I'm considering it future work to relax the double parens requirement.

**Comment by jordwalke** [comment]
Cool, those seem like nice additions too. Is this ready to merge then?

**Comment by jordwalke** [comment]
@kpsuperplane: The Azure builds are getting an error very similar to one you were getting on reason-native:

```
Definition name facebook.reason didn't correspond to a valid definition
```

Do you recall what the culprit is? How do I fix it?

**Comment by kpsuperplane** [comment]
@jordwalke  I solved it last time by renaming the project to something else (e.g. "facebook-reason"), running a successful build, and renaming it back. Never really figured out the root cause though

**Pull request merged**